File: project-euler/Problem_03/problem_3.erl
-module(problem_3).
-compile(export_all).

start()->
    problem_3(600851475143, 2).

problem_3(Num, F) when Num > 1 ->
    case Num rem F of
	0 ->
	    problem_3(Num div F, F);
	_Else ->
	    problem_3(Num, F + 1)
    end;

problem_3(Num, F) ->
    io:format("result: ~p~n", [F]).

File: project-euler/Problem_03/problem_3.go
package main

import (
	"fmt"
)

func problem_3(num int64) int64 {
	var f int64
	f = 2

	for (num > 1) {
		if num % f == 0 {
			num = num / f
		} else {
			f++
		}
	}

	return f
}


func main() {	
	fmt.Println("Result :", problem_3(600851475143))	
}